src/storage/WrappedExpiringLock.ts

```diff
@@ -0,0 +1,69 @@
+import { EventEmitter } from 'events';
+import type { ResourceIdentifier } from '../ldp/representation/ResourceIdentifier';
+import { getLoggerFor } from '../logging/LogUtil';
+import type { ExpiringLock } from './ExpiringLock';
+import type { Lock } from './Lock';
+
+/**
+ * An implementation of an expiring lock which defines the expiration logic.
+ *
+ * ExpiringLock used by a {@link ExpiringResourceLocker} for non-atomic operations.
+ * Emits an "expired" event when internal timer runs out and calls release function when this happen.
+ */
+export class WrappedExpiringLock extends EventEmitter implements ExpiringLock {
+  protected readonly logger = getLoggerFor(this);
+
+  protected readonly innerLock: Lock;
+  protected readonly expiration: number;
+  protected readonly identifier: ResourceIdentifier;
+  protected timeoutHandle?: NodeJS.Timeout;
+
+  /**
+   * @param innerLock - Instance of ResourceLocker to use for acquiring a lock.
+   * @param expiration - Time in ms after which the lock expires.
+   * @param identifier - Identifier of the resource that needs to be locked.
+   */
+  public constructor(innerLock: Lock, expiration: number, identifier: ResourceIdentifier) {
```

Comment:
The `identifier` only gets used for logging so is not needed here. The actual lock on the identifier is going to be the `innerlock`.

Reply:
So completely remove the logging? Or remove the identifier from the message?

Reply:
The logging can stay, just wouldn't add a constructor parameter who's only purpose is being in a logging string.

Reply:
Yes, it could actually

```typescript
this.logger.verbose(`Renewed expiring lock ${this.lock}`);
```

which will call `this.lock.toString()`, so if we implement a good `toString` in `lock`, we have the same info.

Reply:
That would print `Renewed expiring lock lock for <identifier>` with 2 times 'lock', is that okay?

Should I already implement this `toString` method somewhere?
Adding `toString: () => string;` to the Lock interface?

Reply:
No need to add to an interface, it’s standard JavaScript and will be called automatically. Let’s just implement this on the underlying Lock implementation.

Reply:
I think the linter is currently going to complain due to https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-base-to-string.md

Reply:
Yes indeed, linter complains if I do not add that to the interface.

Reply:
@RubenVerborgh's response ended up in the wrong spot, but calling `toString` explicitly makes no difference for the linter rule. So either the rule needs to be disabled or the interface needs to change.

Reply:
👋 bye linter rule

Reply:
@RubenVerborgh removing the linter rule actually does not simply solve the problem, typescript then complains that you can't add a `toString` function to a Lock since the interface does not have that function (typescript error, not linter). Possibly this is because we make the object inline and can be worked around by having a separate class, doing some cast magic, using `Object.create` or something else. But if both the compiler and linter are indicating not to do this we might have to think that there's a point there :D. 

In hindsight, if we do this I am actually in favour of adding it to the Lock interface since otherwise we're probably going to end up in the future with other locks that forget to implement this function.

So this is a lot of words for a simple toString call, so we could also just drop that part 😉

Reply:
So we cannot simply do 

```
this.logger.verbose(`Renewed expiring lock ${this.innerLock}`);
```

and voluntary implement a `toString` method on whatever class `innerLock` is?

Well yeah, let's drop then.

Reply:
Ok, how about we replace https://github.com/solid/community-server/blob/95ab0b4e760107a06a641b86faac7b385a8b1440/src/storage/SingleThreadedResourceLocker.ts#L31-L34

by

```typescript
        resolve({
          release: async(): Promise<void> => {
            this.logger.verbose(`Released ${this}`);
            done();
          },
          toString: (): string => {
            return `lock for ${identifier.path}`;
          },
        } as Lock);
```

Reply:
Okay, nice! So without the `[` `]` around the message?

Reply:
Yeah, that's what I'd propose. But I'll leave the final decision on @joachimvh; we can also just get rid of the logging call, given that the very next line will be the released lock.

Reply:
Ok, final decision 😄 . Since at no point we want to edit interfaces (or constructors) for logging, but we also can't assume Lock is going to have a valid toString function without explicitly defining that: keep the logging in WrappedExpiringLock but remove the identifier reference (so just that it timed out). Since there is equivalent logging in the SingleThreadedLock it will still be clear in the logging which identifier just expired due to the 2 messages being together.